### PR TITLE
Integrate ui-router with dashboardService.current

### PR DESCRIPTION
**Pull request by jmuharsky**

### CODE REVIEW

File: client/components/dashboard/dashboard-service.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ const WidgetType = explorer.models.WidgetType;
 explorer.components.dashboard.DashboardService = function(arrayUtilService,
     errorService, widgetFactoryService, dashboardDataService,
     queryBuilderService,  dashboardVersionService, configService,
-    $filter, $location, $rootScope, $timeout, $window) {
+    explorerStateService, $filter, $location, $rootScope, $timeout, $window) {
```
Comment (klausw) [inline client/components/dashboard/dashboard-service.js]: Arg missing from docstring?

That's a rather long arg list, would be nice to be able to use named args or an initializer dict. OK to leave as is also.

**Comment by klausw** [comment]
LGTM other than nitpick above.

File: client/components/dashboard/dashboard-service.js (hunk 1 above)
Comment (jmuharsky) [inline client/components/dashboard/dashboard-service.js]: I'll add it to the docstring, but the names can't really be reduced as they're used for AngularJS [dependency injection](https://docs.angularjs.org/guide/di); each is the name of an available Angular Service, and AFAIK you can't use an encapsulated object (nor would it necessarily be overall helpful).

Comment (jmuharsky) [inline client/components/dashboard/dashboard-service.js]: The key point here is that user-authored code will never call this constructor.  When the service gets registered, ie: app.Service('myService', MyService) is registered, any parameters on the MyService constructor will be injected as services.

Comment (klausw) [inline client/components/dashboard/dashboard-service.js]: It's ok if this is how it needs to be, it just looks a bit clunky.

**Comment by klausw** [comment]
LGTM

**Comment by jmuharsky** [comment]
Deprecating this PR in favor of #159 (which streamlines the API, adds more tests, and doesn't integrate dashboard.current yet, just adds the supporting classes).

**Pull request closed without merging**